src/through.ml

```diff
@@ -0,0 +1,36 @@
+module Make (C : S.Cache) (DB : S.DB with type key = C.key) = struct
+  type 'a t = 'a C.t
+  type 'a d = 'a DB.d
+  type key = C.key
+
+  let v c = C.v c
+  let stats t = C.stats t
+  let is_empty t = C.is_empty t
+  let capacity t = C.capacity t
+  let size t = C.size t
+  let clear t = C.clear t
+  let mem t d k = C.mem t k || DB.mem d k
+  let find t d k = try C.find t k with Not_found -> DB.find d k
+
+  let find_opt t d k =
+    try Some (C.find t k)
+    with Not_found -> (
+      try
+        let res = DB.find d k in
+        C.replace t k res;
+        Some res
+      with Not_found -> None)
+
+  let promote t d k =
+    (if not (C.mem t k) then
+     try C.replace t k (DB.find d k) with Not_found -> ());
+    C.promote t k
+
+  let replace t d k v =
+    C.replace t k v;
+    DB.replace d k v
+
+  let remove t d k =
+    C.remove t k;
+    DB.remove d k
```

Comment:
What happens if `k` is in db but not in cache?

Reply:
remove in cache will do nothing if k is not find.
```
let remove t k =
    try
      let index, _value = H.find t.tbl k in
      Dllist.remove t.lst index;
      H.remove t.tbl k;
      Stats.remove t.stats
    with Not_found -> ()
```